aten/src/ATen/native/cuda/Copy.cu

```diff
@@ -449,10 +449,10 @@ static void copy_kernel_cuda(TensorIterator& iter, bool non_blocking) {
   }
 
   if (iter.tensor(0).is_conj() != iter.tensor(1).is_conj()) {
-     iter.tensor(0).conj_physical_();
+     iter.tensor(0)._set_conj(iter.tensor(1).is_conj());
```

Comment:
you cannot change `conj` flag on the destination tensor, because the copy is inplace and you are not allowed to change the attributes that the tensor has. THis is the reason `conj_physical` has been run here in the first place - why do you think we would do an operation on tensor data if we didn't need it and could just flip a flag?

Reply:
Alight, misunderstood what `conj_physical` does. Now with the correct understanding of that I don't see anyway for us to ensure that the conjugation happens on the destination after the copy without blocking. It will have to be resolved beforehand. 

We could push this into the branch for copies requiring a temporary, where the conjugate is resolved into a temporary which is then the source for the non blocking copy.

> why do you think we would do an operation on tensor data if we didn't need it and could just flip a flag

FWIW I thought this might be something we don't allow, but there were no test failures, so thought it could be an oversight.

Reply:
I thought there were test cases to check this, but apparently not. We should resolve conj/neg similar to how we resolve dtype via intermediates  - the cases with mismatching conj/neg should go through `requires_temporaries` branch.